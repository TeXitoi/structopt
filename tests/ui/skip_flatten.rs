// Copyright 2018 Guillaume Pinot (@TeXitoi) <texitoi@texitoi.eu>
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use structopt::StructOpt;

#[derive(StructOpt)]
#[structopt(name = "make-cookie")]
struct MakeCookie {
    #[structopt(short)]
    s: String,

    #[structopt(skip, flatten)]
    cmd: Command,
}

#[derive(StructOpt)]
enum Command {
    #[structopt(name = "pound")]
    /// Pound acorns into flour for cookie dough.
    Pound { acorns: u32 },

    Sparkle {
        #[structopt(short)]
        color: String,
    },
}

impl Default for Command {
    fn default() -> Self {
        Pound { acorns: 0 }
    }
}

fn main() {
    let opt = Opt::from_args();
    println!("{:?}", opt);
}
