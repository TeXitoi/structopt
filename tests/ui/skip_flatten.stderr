error: subcommand, flatten and skip cannot be used together
  --> $DIR/skip_flatten.rs:17:23
   |
17 |     #[structopt(skip, flatten)]
   |                       ^^^^^^^
